assign-users.js

```diff
@@ -91,6 +86,7 @@ function matchLearnersToDonors(learners, donations) {
       }
       const cap = calculateLearnerCap(donation);
       console.log('cap is ', cap);
+
       if (!donation.learners || (donation.learners.length < cap)) {
         console.log('found donor: ', donation.sourceDonor);
         foundDonor = true;
```

Comment:
One of the most noticeable bugs I found on the old script was where the matching function failed to move through all possible donations on every user, and we would check the same filled donation every time even on learners that were compatible with other donations. Something to look for. Also need to make sure we've handled the following termination cases:
- ending with extra learners and donations, but no matches
- ending with no learners and open donations
- ending with learners that match the donations, but donations are all closed